test/when_any.cpp

```diff
@@ -252,14 +252,9 @@ TEST_CASE("stop_when is adaptor-pipeable", "[when_any]") {
 }
 
 TEST_CASE("when_any with zero args never completes", "[when_any]") {
-    int value{};
     [[maybe_unused]] auto w = async::when_any();
     static_assert(std::same_as<async::completion_signatures_of_t<decltype(w)>,
                                async::completion_signatures<>>);
-
-    auto op = async::connect(w, receiver{[&] { value = 42; }});
```

Comment:
suggestion: the unit test should still do the connection, just not try to start it. also check the type or properties of the op state.